### ISSUE TITLE
PutObject with streaming results empty Content-Encoding

### ISSUE THREAD

**Issue by trung** [awslabs/aws-crt-java#729]
### Describe the bug

As described in https://github.com/aws/aws-sdk-java-v2/discussions/4642

When using default S3 Async Client, then the issue is no longer there.

### Expected Behavior

No `Content-Encoding` metadata

### Current Behavior

`Content-Encoding` has empty value

### Reproduction Steps

```java
// using CRT
S3AsyncClient s3AsyncClient = S3AsyncClient.crtBuilder()
        .region(awsRegion)
        .credentialsProvider(awsCredentialsProvider)
        .build();

// not using CRT
// S3AsyncClient.builder()
//        .region(awsRegion)
//        .credentialsProvider(awsCredentialsProvider)
//        .build();

Path f = Paths.get("test.pdf");
Flux<ByteBuffer> fileFlux = Flux.using(
    () -> new FileReader(f.toFile()),
    reader -> Flux.fromStream(new BufferedReader(reader).lines()).map(s -> ByteBuffer.wrap(s.getBytes(
        StandardCharsets.UTF_8))),
    reader -> {
      try {
        reader.close();
      } catch (IOException e) {
        throw new RuntimeException(e);
      }
    }
);

final CompletableFuture<PutObjectResponse> s3Future =
        s3AsyncClient.putObject(
            PutObjectRequest.builder()
                .bucket("mybucket")
                .key("test.pdf")
                .contentLength(Files.size(f))
                .contentType(MediaType.APPLICATION_PDF_VALUE)
                .build(),
            AsyncRequestBody.fromPublisher(fileFlux));
```
`test.pdf` has `Content-Encoding` with empty value.
<img width="616" alt="image" src="https://github.com/awslabs/aws-crt-java/assets/24930/dbc7b9e0-e72d-46ad-b0e3-2d2913aef202">

While if using non CRT S3 Async Client, `test.pdf` won't have `Content-Encoding`
<img width="617" alt="image" src="https://github.com/awslabs/aws-crt-java/assets/24930/2d3e3aaf-022b-47ab-8fad-0436c8c2b76d">

### Additional Information/Context

https://github.com/aws/aws-sdk-java-v2/discussions/4642

### aws-crt-java version used

0.28.10

### Java version used

17

### Operating System and version

MacOS 14.1.1 (23B81)

**Comment by jmklix** [comment]
Do you have the logs that show the request for both with and without the CRT?

**Comment by jmklix** [comment]
Can you update to the latest version of the Java sdk? It looks like this was fixed in aws-crt-java version 0.28.11 and the Java sdk updated it's crt version to that this past week

**Comment by trung** [comment]
Hi, updating SDK is not an option for me right now. I decided to use the work around and use non-crt version instead. Will revisit and test this once we update our SDK.
